lib/internal/modules/helpers.js

```diff
@@ -86,20 +86,18 @@ function makeRequireFunction(mod, redirects) {
       if (destination === true) {
         missing = false;
       } else if (destination) {
-        const href = destination.href;
-        if (destination.protocol === 'node:') {
+        const { href, protocol } = destination;
+        if (protocol === 'node:') {
           const specifier = destination.pathname;
 
           if (BuiltinModule.canBeRequiredByUsers(specifier)) {
             const mod = loadBuiltinModule(specifier, href);
             return mod.exports;
           }
           throw new ERR_UNKNOWN_BUILTIN_MODULE(specifier);
-        } else if (destination.protocol === 'file:') {
-          let filepath;
-          if (urlToFileCache.has(href)) {
-            filepath = urlToFileCache.get(href);
-          } else {
+        } else if (protocol === 'file:') {
+          let filepath = urlToFileCache.get(href);
+          if (!filepath) {
```

Comment:
🤔 these are not the same, but I'm pretty sure the difference doesn't matter (an actual value of `filepath` can't be falsy).

Reply:
Yes, the return of `fileURLToPath` will always be truthy.